### PR TITLE
Separate SIKE and SIDH

### PR DESCRIPTION
**Pull request by gadoofou87**
Tried to solve the issue https://github.com/open-quantum-safe/liboqs/issues/737

### CODE REVIEW

**Comment by gadoofou87** [comment]
Also found that the optimized build is broken. Should be fixed in the next commit. Benchmark results:
Before:
```
Speed test
==========
Started at 2020-05-03 03:54:26
Operation                      | Iterations | Total time (s) | Time (us): mean | pop. stdev | CPU cycles: mean          | pop. stdev
------------------------------ | ----------:| --------------:| ---------------:| ----------:| -------------------------:| ----------:
SIDH-p434                      |            |                |                 |            |                           |           
keygen                         |        244 |          3.000 |       12296.766 |     94.246 |                  29510195 |     225677
encaps                         |        122 |          3.020 |       24754.295 |    159.076 |                  59407639 |     379830
decaps                         |        301 |          3.001 |        9969.694 |     92.398 |                  23925222 |     222070
SIDH-p503                      |            |                |                 |            |                           |           
keygen                         |        161 |          3.010 |       18698.199 |    147.583 |                  44873571 |     355170
encaps                         |         81 |          3.036 |       37479.864 |    253.169 |                  89949193 |     607222
decaps                         |        197 |          3.014 |       15297.797 |    111.695 |                  36712490 |     268203
SIDH-p610                      |            |                |                 |            |                           |           
keygen                         |         76 |          3.027 |       39829.526 |    203.560 |                  95587743 |     487759
encaps                         |         42 |          3.053 |       72690.333 |    187.407 |                 174454493 |     455128
decaps                         |         91 |          3.032 |       33322.934 |    259.995 |                  79972195 |     624146
SIDH-p751                      |            |                |                 |            |                           |           
keygen                         |         48 |          3.011 |       62722.542 |    959.625 |                 150531575 |    2302453
encaps                         |         24 |          3.112 |      129678.083 |    393.170 |                 311226119 |     943308
decaps                         |         58 |          3.028 |       52201.052 |   4185.743 |                 125279586 |   10046927
SIDH-p434-compressed           |            |                |                 |            |                           |           
keygen                         |        135 |          3.001 |       22229.941 |   1800.013 |                  53348803 |    4319383
encaps                         |         87 |          3.004 |       34530.322 |    281.753 |                  82870036 |     673612
decaps                         |        272 |          3.009 |       11063.390 |    117.080 |                  26549680 |     280192
SIDH-p503-compressed           |            |                |                 |            |                           |           
keygen                         |         92 |          3.029 |       32927.174 |   3070.929 |                  79021671 |    7369475
encaps                         |         59 |          3.016 |       51116.508 |    519.311 |                 122676888 |    1243204
decaps                         |        180 |          3.010 |       16721.583 |    140.912 |                  40128730 |     336591
SIDH-p610-compressed           |            |                |                 |            |                           |           
keygen                         |         44 |          3.058 |       69492.886 |   5120.177 |                 166779807 |   12287229
encaps                         |         30 |          3.042 |      101391.433 |    846.646 |                 243337144 |    2035569
decaps                         |         82 |          3.030 |       36953.122 |    211.579 |                  88684653 |     506791
SIDH-p751-compressed           |            |                |                 |            |                           |           
keygen                         |         27 |          3.010 |      111466.111 |   9777.405 |                 267514152 |   23465943
encaps                         |         17 |          3.008 |      176921.765 |   1026.379 |                 424609043 |    2462522
decaps                         |         53 |          3.020 |       56989.075 |    357.950 |                 136770951 |     858358
SIKE-p434                      |            |                |                 |            |                           |           
keygen                         |        223 |          3.010 |       13498.888 |    119.820 |                  32394614 |     287604
encaps                         |        135 |          3.010 |       22296.748 |    135.487 |                  53509697 |     323900
decaps                         |        127 |          3.004 |       23652.929 |    156.916 |                  56764577 |     373164
SIKE-p503                      |            |                |                 |            |                           |           
keygen                         |        147 |          3.007 |       20455.170 |    134.236 |                  49089848 |     322668
encaps                         |         89 |          3.028 |       34023.079 |    196.319 |                  81652181 |     470784
decaps                         |         84 |          3.016 |       35909.476 |    563.555 |                  86180297 |    1352840
SIKE-p610                      |            |                |                 |            |                           |           
keygen                         |         77 |          3.034 |       39404.870 |    113.583 |                  94568216 |     271027
encaps                         |         42 |          3.065 |       72975.976 |    171.725 |                 175139156 |     410452
decaps                         |         42 |          3.064 |       72948.238 |    203.847 |                 175073529 |     487798
SIKE-p751                      |            |                |                 |            |                           |           
keygen                         |         44 |          3.063 |       69618.705 |    302.079 |                 167082348 |     725053
encaps                         |         27 |          3.073 |      113807.074 |    444.616 |                 273135076 |    1076387
decaps                         |         25 |          3.031 |      121246.560 |    628.865 |                 290989931 |    1513888
SIKE-p434-compressed           |            |                |                 |            |                           |           
keygen                         |        136 |          3.003 |       22080.699 |   1703.794 |                  52990644 |    4089977
encaps                         |         88 |          3.028 |       34405.886 |    258.290 |                  82571427 |     620387
decaps                         |         93 |          3.002 |       32282.527 |    163.341 |                  77475551 |     392954
SIKE-p503-compressed           |            |                |                 |            |                           |           
keygen                         |         89 |          3.007 |       33791.371 |   3635.482 |                  81095686 |    8721530
encaps                         |         59 |          3.023 |       51240.424 |    492.828 |                 122974711 |    1182452
decaps                         |         62 |          3.017 |       48664.774 |    249.199 |                 116792786 |     597904
SIKE-p610-compressed           |            |                |                 |            |                           |           
keygen                         |         45 |          3.061 |       68031.244 |   4348.805 |                 163272126 |   10436404
encaps                         |         30 |          3.013 |      100423.467 |    860.103 |                 241014069 |    2060175
decaps                         |         31 |          3.087 |       99567.935 |    451.301 |                 238960856 |    1085938
SIKE-p751-compressed           |            |                |                 |            |                           |           
keygen                         |         28 |          3.107 |      110969.857 |   6050.820 |                 266325441 |   14525113
encaps                         |         17 |          3.023 |      177836.765 |    847.103 |                 426806720 |    2031167
decaps                         |         18 |          3.011 |      167291.500 |    655.145 |                 401498079 |    1576518
Ended at 2020-05-03 03:56:51
```
After:
```
Speed test
==========
Started at 2020-05-03 04:06:40
Operation                      | Iterations | Total time (s) | Time (us): mean | pop. stdev | CPU cycles: mean          | pop. stdev
------------------------------ | ----------:| --------------:| ---------------:| ----------:| -------------------------:| ----------:
SIDH-p434                      |            |                |                 |            |                           |           
keygen                         |        257 |          3.003 |       11683.984 |    108.108 |                  28039457 |     258497
encaps                         |        128 |          3.008 |       23503.594 |    198.804 |                  56406442 |     478160
decaps                         |        319 |          3.002 |        9411.285 |     83.354 |                  22584943 |     199559
SIDH-p503                      |            |                |                 |            |                           |           
keygen                         |        171 |          3.015 |       17631.000 |    131.378 |                  42312182 |     314641
encaps                         |         84 |          3.009 |       35826.476 |    228.886 |                  85981081 |     549971
decaps                         |        209 |          3.003 |       14366.244 |    231.801 |                  34476814 |     556085
SIDH-p610                      |            |                |                 |            |                           |           
keygen                         |         84 |          3.033 |       36101.286 |    590.579 |                  86640010 |    1417864
encaps                         |         46 |          3.024 |       65728.326 |    518.795 |                 157745713 |    1244304
decaps                         |        100 |          3.014 |       30142.910 |    293.807 |                  72339877 |     703708
SIDH-p751                      |            |                |                 |            |                           |           
keygen                         |         52 |          3.048 |       58623.750 |    488.726 |                 140693570 |    1171998
encaps                         |         25 |          3.047 |      121887.920 |    467.669 |                 292528149 |    1135340
decaps                         |         63 |          3.039 |       48233.476 |    339.870 |                 115757426 |     818575
SIDH-p434-compressed           |            |                |                 |            |                           |           
keygen                         |        137 |          3.021 |       22052.124 |   2096.253 |                  52921968 |    5031892
encaps                         |         89 |          3.024 |       33978.449 |    330.869 |                  81545535 |     794257
decaps                         |        278 |          3.003 |       10803.011 |    123.997 |                  25924917 |     296844
SIDH-p503-compressed           |            |                |                 |            |                           |           
keygen                         |         94 |          3.025 |       32180.372 |   2590.473 |                  77229449 |    6218219
encaps                         |         61 |          3.023 |       49558.803 |    490.303 |                 118938204 |    1176001
decaps                         |        188 |          3.008 |       16001.947 |    171.277 |                  38402406 |     410703
SIDH-p610-compressed           |            |                |                 |            |                           |           
keygen                         |         49 |          3.014 |       61515.510 |   4431.950 |                 147633868 |   10631421
encaps                         |         33 |          3.041 |       92139.455 |    958.592 |                 221132538 |    2308318
decaps                         |         90 |          3.002 |       33358.289 |    419.392 |                  80057050 |    1007012
SIDH-p751-compressed           |            |                |                 |            |                           |           
keygen                         |         30 |          3.112 |      103734.967 |   7567.334 |                 248961419 |   18164686
encaps                         |         19 |          3.164 |      166539.474 |   1456.140 |                 399692220 |    3500600
decaps                         |         57 |          3.015 |       52894.368 |    316.876 |                 126943793 |     755903
SIKE-p434                      |            |                |                 |            |                           |           
keygen                         |        234 |          3.008 |       12854.192 |    152.421 |                  30847848 |     365191
encaps                         |        142 |          3.010 |       21196.669 |    321.751 |                  50869934 |     770682
decaps                         |        135 |          3.017 |       22347.333 |    164.975 |                  53630971 |     397143
SIKE-p503                      |            |                |                 |            |                           |           
keygen                         |        155 |          3.008 |       19408.852 |    109.125 |                  46578648 |     261675
encaps                         |         94 |          3.018 |       32101.319 |    267.101 |                  77040964 |     638110
decaps                         |         89 |          3.032 |       34068.270 |    211.372 |                  81760881 |     508299
SIKE-p610                      |            |                |                 |            |                           |           
keygen                         |         84 |          3.019 |       35941.774 |    194.536 |                  86257192 |     465098
encaps                         |         46 |          3.042 |       66127.978 |    359.612 |                 158704336 |     866516
decaps                         |         45 |          3.014 |       66974.822 |    372.540 |                 160736848 |     898663
SIKE-p751                      |            |                |                 |            |                           |           
keygen                         |         46 |          3.011 |       65464.304 |    400.569 |                 157111379 |     962019
encaps                         |         29 |          3.080 |      106192.414 |    512.519 |                 254859512 |    1230130
decaps                         |         27 |          3.085 |      114242.889 |    291.898 |                 274180089 |     704248
SIKE-p434-compressed           |            |                |                 |            |                           |           
keygen                         |        137 |          3.022 |       22057.642 |   2126.419 |                  52936087 |    5103614
encaps                         |         89 |          3.026 |       33995.551 |    368.750 |                  81586802 |     885006
decaps                         |         95 |          3.028 |       31872.263 |    228.079 |                  76490696 |     546417
SIKE-p503-compressed           |            |                |                 |            |                           |           
keygen                         |         94 |          3.000 |       31918.979 |   3090.428 |                  76602061 |    7417462
encaps                         |         61 |          3.005 |       49267.508 |    555.287 |                 118239558 |    1331580
decaps                         |         65 |          3.014 |       46371.754 |    296.945 |                 111289174 |     713270
SIKE-p610-compressed           |            |                |                 |            |                           |           
keygen                         |         48 |          3.033 |       63180.729 |   6154.172 |                 151631028 |   14771243
encaps                         |         33 |          3.034 |       91934.848 |    579.692 |                 220640360 |    1386825
decaps                         |         34 |          3.049 |       89687.735 |    992.487 |                 215248482 |    2384378
SIKE-p751-compressed           |            |                |                 |            |                           |           
keygen                         |         30 |          3.063 |      102104.900 |   6531.094 |                 245046051 |   15669453
encaps                         |         19 |          3.130 |      164737.158 |   1208.861 |                 395366767 |    2896534
decaps                         |         20 |          3.091 |      154560.400 |    384.045 |                 370943455 |     928996
Ended at 2020-05-03 04:09:06
```

**Comment by christianpaquin** [comment]
I'll take a look, most likely on Wednesday

**Comment by christianpaquin** [comment]
We are currently in a code freeze to snapshot OQS. SIKE will be updated to [v3.3](https://github.com/microsoft/PQCrypto-SIDH/tree/v3.3) as soon as this is done. I'll take this PR into consideration when updating SIDH/SIKE.

File: src/kem/sike/CMakeLists.txt [hunk 1]
```diff
@@ -23,22 +23,27 @@ if(WIN32 AND NOT (MINGW OR MSYS OR CYGWIN))
...
+if(OQS_ENABLE_KEM_SIDH)
+    set(SRCS ${SRCS} kem_sidh.c)
+endif()
 
 if(ARCH STREQUAL "x86_64")
     if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
         set(SRCS ${SRCS} P503/AMD64/fp_x64_asm.S
                          P751/AMD64/fp_x64_asm.S)
-        if(USE_BMI2_INSTRUCTIONS AND NOT OQS_PORTABLE_BUILD)
+        if(OQS_USE_BMI2_INSTRUCTIONS AND NOT OQS_PORTABLE_BUILD)
```
Comment (xvzcf) [inline src/kem/sike/CMakeLists.txt:44]: I opened another PR with this particular fix, since it would be worth having for the release.

Comment (gadoofou87) [reply src/kem/sike/CMakeLists.txt]: Everything's fine. I removed that code from my PR

**Comment by christianpaquin** [comment]
This should now be a duplicate of PR #769. @gadoofou87, let me know if I missed something.

**Comment by xvzcf** [comment]
Closing this for now. @gadoofou87 feel free to re-open this PR/open a new PR if there's something missing.

**Pull request closed without merging**